### PR TITLE
Update dependencies, translations; add Catalan

### LINKED ISSUE #310: Create Catalan translation
Hi!
Catalan translation is done in Transifex so, can you create that?

### LINKED ISSUE #311: Upgrade to Transifex API v3
This repository currently uses Transifex API v2, which was sunsetted over a year ago. In order to pull in anything from Transifex, it needs to be upgraded to use Transifex API v3, one of the compatible SDKs, or the `tx` command line tool.

/ref openstreetmap/iD#9375

### PR DESCRIPTION
**Pull request by 1ec5**
Replaced manual HTTP calls to the defunct Transifex API v2 with the Transifex API v3 JavaScript SDK. This required increasing the minimum Node version from 8 to 16 and rewriting the transifex.js script to use `async`/`await` and promises.

Added a complete Catalan localization from Transifex. Reran `npm run transifex` to synchronize this repository with Transifex for the first time since December 2019. I kept various pre-translations where an English string would otherwise have been backfilled. Thanks to the following translators for their help over the past four years or so, and apologies for the delay in pulling in the changes:

* @eduardoaddad for Brazilian Portuguese
* @rroset for Catalan
* @cmahopper for Danish
* [arjanTransfix](https://app.transifex.com/user/profile/arjanTransfix/) for Dutch
* @verdulo for Esperanto
* @operon for French
* @debyos for Hungarian
* [nob.nihashi](https://app.transifex.com/user/profile/nob.nihashi/) for Japanese
* [rafal.ferenc.broomee](https://app.transifex.com/user/profile/rafal.ferenc.broomee/) for Polish
* @yuryleb for Russian
* @chriswu42 for Simplified Chinese
* @riiga for Swedish
* @andygol for Ukrainian
* @toblinkz for Yoruba

Fixes #310 and fixes #311. Supersedes #292, #297, #306, and #307.

### CODE REVIEW

**Review by freenerd — APPROVED** [review]
@1ec5 ❤️❤️❤️ lgtm

Are you going to release v0.15.0?

File: scripts/transifex.js [hunk 1]
```diff
@@ -1,44 +1,57 @@
...
-// but needs authentication via username and password
-// http://docs.transifex.com/api/#authentication
+// but needs authentication via username and bearer token
+// https://developers.transifex.com/reference/api-authentication
 //
 // auth is stored in ./transifex.auth as json object
 //  {
-//      "user": "username",
-//      "pass": "password"
+//      "token": "1/0b4e75552e481aeae813aebad53f3de3fcc67ccd"
```
[the remark this answers is not in the document]
Comment (1ec5) [reply scripts/transifex.js:16]: It’s the one Transifex hard-coded into their documentation. It isn’t mine, in any event. 😅

**Comment by 1ec5** [comment]
> Are you going to release v0.15.0?

Sure, if I can still figure out how to work this thing.

**Comment by 1ec5** [comment]
Tracking the release in #313.

**Pull request merged**